### PR TITLE
use go-gitlog as an alternative to git2go

### PR DESCRIPTION
**Pull request by matthyx**
@dwertent we should check if this is acceptable, since we require a `git` binary in the PATH
(probably OK to assume if you want to scan a git repo, you should have git installed)

benchmark shows we allocate more memory, but we're faster:
```
gitlog
BenchmarkBuildCommitMap-8   	       5	 206852705 ns/op	13889608 B/op	   87819 allocs/op
git2go
BenchmarkBuildCommitMap-8   	       4	 310221194 ns/op	 3428458 B/op	  144074 allocs/op
```

### CODE REVIEW

**Comment by matthyx** [comment]
S_L_O_W again ... but this is because we don't have memoization - I will try to add it when I have some time

**Comment by matthyx** [comment]
cc @fredbi

File: core/cautils/git_native_disabled.go [hunk 1]
```diff
@@ -3,20 +3,85 @@
...
+	}
+
+	if relevantCommit, exists := g.fileToLastCommit[filePath]; exists {
+		return g.getCommit(relevantCommit), nil
+	}
+
+	return nil, fmt.Errorf("failed to get commit information for file: %s", filePath)
+}
+
+func (g *gitRepository) buildCommitMap() {
```
Comment (dwertent) [inline core/cautils/git_native_disabled.go:39]: There is no need for `//go:build !gitenabled` anymore. So you do not need to duplicate the code in both files.
Also, we can update the installation and remove the git2lib from the installation (and use `git build` instead)

**Review by dwertent — COMMENTED** [review]
Please see my comment, WDYT?

File: core/cautils/git_native_disabled.go (hunk 1 above)
Comment (matthyx) [reply core/cautils/git_native_disabled.go:39]: yes, but maybe for the moment we can build both and make sure it works for all users

**Comment by dwertent** [comment]
Waiting for PR checks to pass

**Comment by dwertent** [comment]
I think we can remove logs like this:
```
 ❗ Git scan skipped. error: failed to get commit information for file: bla.yaml
```
Maybe change it to debug

**Comment by matthyx** [comment]
> Maybe change it to debug

Done

**Comment by matthyx** [comment]
it now works on Windows, issue was path normalization when accessing commits in the `fileToLastCommit` map

**Comment by craigbox** [comment]
Closes #1245.

**Comment by matthyx** [comment]
fixes https://github.com/kubescape/kubescape/issues/832

**Review by dwertent — DISMISSED** [review]

**Review by dwertent — APPROVED** [review]

**Pull request merged**